lib/Jsrt/JsrtDebugUtils.cpp

```diff
@@ -0,0 +1,465 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "JsrtPch.h"
+#include "JsrtDebugUtils.h"
+#include "RuntimeDebugPch.h"
+#include "screrror.h"   // For CompileScriptException
+
+void JsrtDebugUtils::AddScriptIdToObject(Js::DynamicObject* object, Js::Utf8SourceInfo* utf8SourceInfo)
+{
+    JsrtDebugUtils::AddPropertyToObject(object, JsrtDebugPropertyId::scriptId, utf8SourceInfo->GetSourceInfoId(), utf8SourceInfo->GetScriptContext());
+
+    Js::Utf8SourceInfo* callerUtf8SourceInfo = utf8SourceInfo->GetCallerUtf8SourceInfo();
+    if (callerUtf8SourceInfo != nullptr)
+    {
+        JsrtDebugUtils::AddPropertyToObject(object, JsrtDebugPropertyId::parentScriptId, callerUtf8SourceInfo->GetSourceInfoId(), callerUtf8SourceInfo->GetScriptContext());
+    }
+}
+
+void JsrtDebugUtils::AddFileNameOrScriptTypeToObject(Js::DynamicObject* object, Js::Utf8SourceInfo* utf8SourceInfo)
+{
+    if (utf8SourceInfo->IsDynamic())
+    {
+        AssertMsg(utf8SourceInfo->GetSourceContextInfo()->url == nullptr, "How come dynamic code have a url?");
```

Comment:
ok - but I don't see a need of assert here. All we are looking for GetSourceName, and whatever it gives.


---
In reply to: [62749710](https://github.com/Microsoft/ChakraCore/pull/926#discussion_r62749710) [](ancestors = 62749710)